OpenSim/Common/Object.h

```diff
@@ -554,6 +556,13 @@ class OSIMCOMMON_API Object
         _objectIsUpToDate = false;
     }
 
+    /** To be up-to-date with properties we must make sure the name of an object,
+    which can be changed independent of the property name, remains consistent
+    with any restrictions specified by the Property. For example, unnamed
+    properties should not have a name. Furthermore, named properties whose object
+    has no name, should have the property name. **/
+    void makeObjectNamesConsistentWithProperties();
```

Comment:
I think we should document that this function is recursive (it visits each node of the ownership tree).